probe-rs/src/architecture/riscv/sequences/esp32h2.rs

```diff
@@ -0,0 +1,55 @@
+//! Sequences for the ESP32H2.
+
+use std::sync::Arc;
+
+use super::RiscvDebugSequence;
+use crate::{
+    architecture::riscv::communication_interface::RiscvCommunicationInterface, MemoryInterface,
+};
+
+/// The debug sequence implementation for the ESP32H2.
+#[derive(Debug)]
+pub struct ESP32H2(());
+
+impl ESP32H2 {
+    /// Creates a new debug sequence handle for the ESP32H2.
+    pub fn create() -> Arc<dyn RiscvDebugSequence> {
+        Arc::new(Self(()))
+    }
+}
+
+impl RiscvDebugSequence for ESP32H2 {
+    fn on_connect(&self, interface: &mut RiscvCommunicationInterface) -> Result<(), crate::Error> {
+        tracing::info!("Disabling esp32h2 watchdogs...");
+        // disable super wdt
+        interface.write_word_32(0x600B1C20, 0x50D83AA1)?; // write protection off
+        let current = interface.read_word_32(0x600B_1C1C)?;
+        interface.write_word_32(0x600B_1C1C, current | 1 << 18)?; // set RTC_CNTL_SWD_AUTO_FEED_EN
+        interface.write_word_32(0x600B1C20, 0x0)?; // write protection on
+
+        // tg0 wdg
+        interface.write_word_32(0x6000_8064, 0x50D83AA1)?; // write protection off
+        interface.write_word_32(0x6000_8048, 0x0)?;
+        interface.write_word_32(0x6000_8064, 0x0)?; // write protection on
+
+        // tg1 wdg
+        interface.write_word_32(0x6000_9064, 0x50D83AA1)?; // write protection off
+        interface.write_word_32(0x6000_9048, 0x0)?;
+        interface.write_word_32(0x6000_9064, 0x0)?; // write protection on
+
+        // rtc wdg
+        interface.write_word_32(0x600B_1C18, 0x50D83AA1)?; // write protection off
+        interface.write_word_32(0x600B_1C00, 0x0)?;
+        interface.write_word_32(0x600B_1C18, 0x0)?; // write protection on
+
+        Ok(())
+    }
+
+    fn detect_flash_size(
+        &self,
+        interface: &mut RiscvCommunicationInterface,
+    ) -> Result<Option<usize>, crate::Error> {
+        super::esp32_attach_flash(interface, 0x40820000, 0x400001d4)?;
```

Comment:
Some brief comments about what these addresses are, and if they're ROM functions, their exported name. I know this can be inferred from the arg name in the function but its nicer if its here and perhaps different chips might need different from functions in the future.